### ISSUE TITLE
2.6.0 : Properties are not read anymore (regression)

### ISSUE THREAD

**Issue by cbornet** [springfox/springfox#1514]
- [x] What version of the library are you using? **2.6.0**
- [x] Bug report. 
    - [x] if you have a repo that demonstrates the issue for bonus points! **https://github.com/cbornet/springfox-demos**

Values from property file are not read when including @EnableSwagger2
I've added a /test endpoint in [this commit](
https://github.com/cbornet/springfox-demos/commit/f88451d911044026cd8f06ded63697171937fa44) to show the issue
```
curl "http://localhost:8081/springfox/test"
8080
```
Response should be 8081.

Downgrading to 2.5.0 solves the issue.

Probably linked to https://github.com/springfox/springfox/commit/e545b61e46513c131a009bf82ab2879ca4cc58d0

**Comment by dilipkrish** [comment]
Yeah, this is a tricky/known issue. Reverted to using `PropertyPlaceholderConfigurer` to get around this problem. See [here](https://github.com/springfox/springfox/issues/1414) for details.

**Comment by dilipkrish** [comment]
May be worth updating the docs

**Comment by cbornet** [comment]
I don't understand : it's a bug ! It should be fixed !

**Comment by cbornet** [comment]
See https://github.com/jhipster/generator-jhipster/issues/4283. Because of this, we can't upgrade jhipstzr to springfox 2.6.0

**Comment by dilipkrish** [comment]
Let me take a look... in the mean while let me know if you have any ideas

**Comment by dilipkrish** [comment]
@cbornet this fixes it.. but its not ideal. Let me know if you have better ideas

**Comment by dilipkrish** [comment]
@cbornet could you verify the fix in 2.6.1-SNAPSHOT. If everything looks good I'll release

**Comment by cbornet** [comment]
@dilipkrish It works like a charm. Thanks for your quick answer !
I'll upgrade to 2.6.1 as soon a it's released.

**Comment by dilipkrish** [comment]
Woot! I'll wait a couple weeks for more bug fixes and feedback and release it!

**Comment by Zhuinden** [comment]
@dilipkrish Guys it's been 2 weeks

**Comment by macarr** [comment]
I'm having the same issue with 2.6.1

for the following code:
@Value("${oracle.url}")
    private String url;

Using 2.5.0 - resolves to the value in the properties file
Using 2.6.1 - resolves to ${oracle.url}

**Comment by dilipkrish** [comment]
Are you using spring boot? Could you provide a sample, because this appears to be fixed. Also create a new issue if you find that its truly still broken

**Comment by macarr** [comment]
No, I'm running a war file on a wildfly 10 server.

**Comment by dilipkrish** [comment]
I have no idea It may be an artifact of spring, wildfly and your environment.

**Comment by Verdoso** [comment]
Hi,
We are having the same issue. We tried to migrate to 2.6.1 from 2.4.0 to solve an issue with xml/json options in the interface, and 2.6.x versions cause all config server based properties to stop working. We tried 2.5.0, 2.6.0, and 2.6.1 and it just happens with 2.6.x versions.
This app runs as a SpringBoot executable jar (container embedded).

**Comment by Verdoso** [comment]
Ok,
Good news. I traced down our issue and it seems that the current behaviour is fine, whereas up to version 2.5.0 Swagger was masking some errors we had in our code.
We had some properties defines as such:

    @Value("${defaults.whatever}")
    private String defaultWhatever;

This placeholder has no default value, so it means the property is mandatory. Up to 2.5.0, I guess Swagger was doing something and providing a default value for those properties. From 2.6.X, Swagger does not provide a default value and SpringBoot complains that it cannot resolve ${defaults.whatever}.

So, in this case, it was a misconfiguration in our side that had been masked. Once we specified that the properties are optional with the colon at the end, everything worked.

    @Value("${defaults.whatever:}")

Cheers!
